packages/@webex/internal-plugin-task/src/task.js

```diff
@@ -0,0 +1,241 @@
+/*!
+ * Copyright (c) 2015-2020 Cisco Systems, Inc. See LICENSE file.
+ */
+import {isArray} from 'lodash';
+import {WebexPlugin} from '@webex/webex-core';
+
+import {RAINDROP_REGISTERED, RAINDROP_UNREGISTERED} from './constants';
+
+import EncryptHelper from './encrypt.helper';
+import DecryptHelper from './decrypt.helper';
+
+const Task = WebexPlugin.extend({
+  namespace: 'Task',
+
+  /**
+   * registered value indicating events registration is successful
+   * @instance
+   * @type {Boolean}
+   * @memberof Task
+   */
+  registered: false,
+
+  /**
+   * Cache KMS encryptionKeyUrl
+   * */
+  encryptionKeyUrl: null,
+
+  /**
+   * WebexPlugin initialize method. This triggers once Webex has completed its
+   * initialization workflow.
+   *
+   * If the plugin is meant to perform startup actions, place them in this
+   * `initialize()` method instead of the `constructor()` method.
+   * @returns {void}
+   */
+  initialize() {
+    // Used to perform actions after webex is fully qualified and ready for
+    // operation.
+    this.listenToOnce(this.webex, 'ready', () => {
+      // Pre-fetch a KMS encryption key url to improve performance
+      this.webex.internal.encryption.kms.createUnboundKeys({count: 1}).then((keys) => {
+        const key = isArray(keys) ? keys[0] : keys;
+        this.encryptionKeyUrl = key ? key.uri : null;
+        this.logger.info('Task->bind a KMS encryption key url');
+        this.webex.internal.encryption
+          .getKey(this.encryptionKeyUrl, {onBehalfOf: null})
+          .then((retrievedKey) => {
+            this.encryptionKeyUrl = retrievedKey ? retrievedKey.uri : null;
+            this.logger.info('Task->retrieve the KMS encryption key url and cache it');
+          });
+      });
+    });
+  },
+
+  /**
+   * Explicitly sets up the Task plugin by registering
+   * the device, connecting to mercury, and listening for events.
+   * @returns {Promise}
+   * @public
+   * @memberof Task
+   */
+  register() {
+    if (!this.webex.canAuthorize) {
+      this.logger.error('Task->register#ERROR, Unable to register, SDK cannot authorize');
+
+      return Promise.reject(new Error('SDK cannot authorize'));
+    }
+
+    if (this.registered) {
+      this.logger.info('Task->register#INFO, Calendar plugin already registered');
+
+      return Promise.resolve();
+    }
+
+    return this.webex.internal.device
+      .register()
+      .then(() => this.webex.internal.mercury.connect())
+      .then(() => {
+        this.listenForEvents();
+        this.trigger(RAINDROP_REGISTERED);
+        this.registered = true;
+      })
+      .catch((error) => {
+        this.logger.error(`Task->register#ERROR, Unable to register, ${error.message}`);
+
+        return Promise.reject(error);
+      });
+  },
+
+  /**
+   * Explicitly tears down the calendar plugin by de-registering
```

Comment:
are these bad copy/pastes in the comments? i see lots of references to calendar plugin in this file in the comments. i don't think that's correct?